ts/Extensions/Annotations/NavigationBindings.ts

```diff
@@ -1602,27 +1602,30 @@ addEvent(Chart, 'render', function (): void {
 
                     if (buttonNode) {
                         if (value.noDataState === 'normal') {
-                            buttonNode.forEach(function (button): void {
+                            for (let i = 0; i < buttonNode.length; i++) {
+                                const button = buttonNode[i];
                                 // If button has noDataState: 'normal',
                                 // and has disabledClassName,
                                 // remove this className.
                                 if (button.className.indexOf(disabledClassName) !== -1) {
                                     button.classList.remove(disabledClassName);
                                 }
-                            });
+                            }
                         } else if (!buttonsEnabled) {
-                            buttonNode.forEach(function (button): void {
+                            for (let i = 0; i < buttonNode.length; i++) {
+                                const button = buttonNode[i];
                                 if (button.className.indexOf(disabledClassName) === -1) {
                                     button.className += ' ' + disabledClassName;
                                 }
-                            });
+                            }
                         } else {
-                            buttonNode.forEach(function (button): void {
+                            for (let i = 0; i < buttonNode.length; i++) {
+                                const button = buttonNode[i];
```

Comment:
What about simply convert a nodeList to an array like this?
```js
const buttonNode = Array.prototype.slice.call(
    chart.navigationBindings.container[0].querySelectorAll('.' + key)
);
```

Reply:
I moved the for loop outside the ifs instead.